docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1061 @@
+
+![Data Structure Class Diagram](./static/data/DataStructureClassDiagram.png)
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+## Communication
+
+This project uses a simplified HTTP model, where the frontend sends a `Request` to the backend to perform data-related operations. The backend returns a `Response`, which is then processed by the frontend
+
+![Simplified HTTP Model](static/communication/SimplifiedHTTPClassDiagram.png)
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+<!-- @@author -->
+
+<!-- # Implementation -->
+
+<!-- ## [Proposed] Undo/Redo feature -->
+
+# Testing
+
+## Unit Tests
+We adopt the Arrange, Act, Assert pattern for unit tests in this project.
+This allows us to achieve a structured unit tests while balancing code readability and maintainability, and allowing a clear separation of the setup, operations and results.
+For backend testing, we use utility classes such as `EntryTestUtil` and `BackendTestUtil` to reduce code repetition and to simplify the testing process.
+
+<details>
+<summary>Example</summary>
+
+  ```java
+  @DisplayName("Test /entries [GET]")
+class TestEntriesGet extends EntryTestUtil {
+    private static final EntryLog expectedEntryLog = new EntryLog();
+
+    @BeforeEach
+    void init() {
+        TEST_BACKEND.clearData();
+        expectedEntryLog.clearAllEntries();
+    }
+
+    @Test
+    void entriesEndpointGET_recentEntries_correctEntries() {
+        // Arrange
+        addEntry(ENTRY_1);
+        addEntry(ENTRY_2);
+        addEntry(ENTRY_3);
+        addEntry(ENTRY_4);
+        expectedEntryLog.addEntry(ENTRY_3);
+        expectedEntryLog.addEntry(ENTRY_4);
+
+        // Act
+        Request request = new Request(RequestMethod.GET); // recent 2 entries
+        request.addParam(RequestParams.NUM_ENTRIES, "2");
+        Response response = TEST_BACKEND.requestEndpointEntries(request);
+        EntryLog returnedEntryLog = EntryLogParser.deserialise(response.getData());
+
+        // Assert
+        assertEquals(response.getResponseStatus(), ResponseStatus.OK);
+        assertTrue(isSameEntryLog(expectedEntryLog, returnedEntryLog));
+    }
+}
+  ```
+
+  </details>
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+<!-- ## Integration Testing
+
+Integration testing in our application is performed by testing `Frontend` and `Backend`, where we ensure that the relevant features 
+for each major component is working, before they are combined and tested through system testing
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div> -->
+
+<!-- ## System Testing
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div> -->
+
+<!-- @@author adenteo -->
+## Instructions for manual testing
```

Comment:
Formatting of all the test case outputs are incorrect and does not show up properly on the DG. Do check your markdowns.
<img width="1056" alt="Screenshot 2023-03-30 at 1 43 17 PM" src="https://user-images.githubusercontent.com/88520987/228740615-ccbcc2ca-1a2c-401c-8483-299b588fc3dd.png">